tests/processes/test_wps_package.py

```diff
@@ -667,3 +668,84 @@ def test_mask_process_inputs(inputs, expect):
 def test_format_extension_validator_basic(data_input, mode, expect):
     # type: (Any, int, bool) -> None
     assert format_extension_validator(data_input, mode) == expect
+
+
+@pytest.mark.parametrize("original, expected", [
+    (
+        # Test author metadata with empty wps_package
+        {
+            "cwl_package_package": {
+                "s:author": [
+                    {"class": "s:Person", "s:name": "John Doe", "s:affiliation": "Example Inc."}
+                ],
+
+            },
+            "wps_package_metadata": {}
+        },
+        {
+            "metadata": [
+                {
+                    "role": "author",
+                    "value": {
+                        "$schema": "https://schema.org/Person",
+                        "name": "John Doe",
+                        "affiliation": "Example Inc."
+                    }
+                }
+            ]
+        }
+    ),
+    (
+        # Test codeRepository
+        {
+            "cwl_package_package": {
+                "s:codeRepository": "https://gitlab.com/",
+            },
+            "wps_package_metadata": {}
+        },
+        {
+            "metadata": [
+                {
+                    "rel": "codeRepository",
+                    "href": "https://gitlab.com/"
```

Comment:
For this case, I think it is safe to assume `type: text/html` could be used. You can set it by default for `codeRepository`. 

Consider adding a `s:license` test case as well. In that case, you can use
`weaver.format.get_content_type(os.path.splitext(<URL>)[-1], default=ContentType.TEXT_PLAIN)` to resolve the specified `type`.
A `https://example.com/LICENSE.md` should result in `text/markdown`.

Reply:
So the type should be added to the metadata when it's a link ? And if its a codeRepository we can always assume the html type without using `weaver.format.get_content_type(os.path.splitext(<URL>)[-1], default=ContentType.TEXT_PLAIN)`
```
{
    "type": "type: text/html"
    "rel": "codeRepository",
    "href": "https://gitlab.com/"
}
```

Reply:
`"type": "text/html"`
And yes, it can be assumed to simplify things.